### PR TITLE
multiple-model-recipients

### PR DESCRIPTION
**Pull request by musonza**

### CODE REVIEW

**Comment by musonza** [comment]
@SimonJ did these updates work for your use case?

**Comment by SimonJ** [comment]
The tests works fine but i can't run the migration file with `php artisan migrate` 🤔

**Comment by musonza** [comment]
This would be a major update which requires to drop the existing chat
tables. Is this in development or you can’t afford to do that?

**Comment by SimonJ** [comment]
@musonza it's in develop so i can just refresh the migrations but it errors out as described in the issue :)

**Comment by SimonJ** [comment]
https://github.com/musonza/chat/issues/157

**Comment by SimonJ** [comment]
Hey mate :) did you have time to check on this @musonza

**Comment by musonza** [comment]
hi @SimonJ sorry i haven't had the chance. Can you link the original error? I don't see it

**Comment by SimonJ** [comment]
Hi @musonza You can see it here https://github.com/musonza/chat/issues/157

File: app/Http/Controllers/ConversationController.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ public function index()
 
     public function store()
     {
-        $participants = [auth()->user()->id];
+        $participants = [auth()->user()];
```
Comment (SimonJ) [inline app/Http/Controllers/ConversationController.php]: We also need the user that it wants to have a conversation with, right?

**Comment by SimonJ** [comment]
@musonza I think the `user_id` on `mc_messages` also needs to be changed

File: app/Http/Controllers/ConversationController.php (hunk 1 above)
Comment (musonza) [reply app/Http/Controllers/ConversationController.php]: So this is just for demo purpose. Just like when you start a slack channel you get added to the conversation and then you can use the `join` function to add additional users

Comment (SimonJ) [reply app/Http/Controllers/ConversationController.php]: Ahh, make sense :)

Comment (SimonJ) [reply app/Http/Controllers/ConversationController.php]: The way im doing it ATM. is to check if the chat is started from the client table or the user table, like this

```php
    public function store()
    {
        $participant_id = request()->participant_id;
        $participant_class = request()->participant_class;

        if ($participant_class === 'client') {
            $participant = Client::find($participant_id);
        } else {
            $participant = User::find($participant_id);
        }

        $participants = [auth()->user(), $participant];
        $conversation = Chat::createConversation($participants);

        return response($conversation);
    }
```

Then the frontend send a request with the id and the class name

Comment (musonza) [reply app/Http/Controllers/ConversationController.php]: Yes that makes sense. If you have a bunch of participant types you may want to pass the full a fully qualified namespace for `participant_class`

```php
    public function store()
    {
        $participantId = request()->participant_id;
        $participantClass = request()->participant_class;
        
        $participants = [auth()->user()]

        if ($participantClass) {
           // An exception will be thrown for invalid classes or you can add validation for allowed classes
           $participants[] = app($participantClass)->find($participantId);
        }

        $conversation = Chat::createConversation($participants);

        return response($conversation);
    }
```

**Pull request merged**